modules/page-memory/src/test/java/org/apache/ignite/internal/pagememory/persistence/store/AbstractFilePageStoreIoTest.java

```diff
@@ -373,4 +376,36 @@ void testRenameAndEnsureRace() throws Exception {
             verify(ioFactory).create(newFilePath, CREATE, READ, WRITE);
         }
     }
+
+    @ParameterizedTest
+    @MethodSource("ioFactories")
+    void testRenameAndReadRace(FileIoFactory ioFactory) throws Exception {
+        Path filePath = workDir.resolve("test");
+
+        try (AbstractFilePageStoreIo filePageStoreIo = createFilePageStoreIo(filePath, ioFactory)) {
+            filePageStoreIo.ensure();
+
+            long expPageId = createDataPageId(() -> 1);
+
+            ByteBuffer byteBuffer = createPageByteBuffer(expPageId, PAGE_SIZE);
+
+            filePageStoreIo.write(expPageId, byteBuffer.rewind(), true);
+
+            // Loop works way better when you need to reproduce a particularly naughty race.
```

Comment:
Better than what? Probably better than `@RepeatedTest`, but it's not clear from the context. Let's either clarify the comment or drop it.

Reply:
Ill clarify it